cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt

```diff
@@ -181,43 +181,43 @@ class ExpressionHandler(frontend: PythonLanguageFrontend) :
         return subscriptExpression
     }
 
+    /**
+     * Handles a [Python.AST.BoolOp]. This can either be a chain of `and` or `or` expressions. We
+     * model this as [BinaryOperator]s with nesting. Thus, `a and b and c` becomes `a and (b and c)`
+     * (same for `or`).
+     */
     private fun handleBoolOp(node: Python.AST.BoolOp): Expression {
         val op =
             when (node.op) {
                 is Python.AST.And -> "and"
                 is Python.AST.Or -> "or"
             }
-        return handleBoolOpRecursively(node.values, op, node)
-    }
+        return when (node.values.size) {
+            0,
+            1 ->
+                newProblemExpression(
+                    "handleBoolOp: Expected at least two expressions but got " + node.values.size,
+                    rawNode = node
+                )
+            else -> {
+                val result = newBinaryOperator(operatorCode = op, rawNode = node)
+                var idx = 0
+                var currentInnerNode: BinaryOperator = result
+
+                // For all iterations until the last one: set the lhs to handle the current idx and
+                // create a new nested [BinaryOperator] for the rhs. Then slide the window by one.
+                while (idx < node.values.size - 2) {
```

Comment:
More readable would actually be a for loop until:
`for(idx in 0 until node.values.size - 2)`

Reply:
But I also need the `idx` after the loop. Will change to this, if you prefer this style.

Reply:
aaah, got it. in this case you cant use for